### ISSUE TITLE
Add tools for GDPR compliance 🇪🇺

### ISSUE THREAD

**Issue by pferreir** [indico/indico#1415]
``` yaml
# Ticket imported from Trac
```

There is a new CERN Data Protection Policy being drafted. This will imply that all CERN-stored data will have to respect a series of constraints, depending on its classification. Indico already does something like that, as content can be set as "public" or "restricted". However, other levels may be required.

Since  we will have to eventually conform to this policy at CERN, it would be nice to find a simple and configurable scheme that allows administrators to cope with local security rules without necessarily enforcing any particular behavior.

**Comment by TimSmithCH** [comment]
GDPR will arrive in May 2018, and CERN's version (OC11) about the same time, so we should chart out minimum required actions:
- Add a privacy statement hook in the footer of all pages, next to T&C
- Add a default privacy statement which should include "details on how the data will be processed, who it may be passed to and how long it will be kept for. It also informs the individual of how to exercise their rights as a data subject"
- Add a "conspicuous link" next to the "apply" button on all registration forms, pointing to the data privacy statement, and saying that by proceeding they acknowledge and accept
- Ensure personal registration data (e.g DoB, ID card numbers, etc) are can be deleted automatically after (configuration parameter) specified number of weeks after the conference
- more?

**Comment by pferreir** [comment]
@TimSmithCH Someone mentioned cookies recently. Do we have to add one of those "annoying" "this site uses cookies" warnings?

**Comment by bpedersen2** [comment]
Related discussion on is here as well: [](https://talk.getindico.io/t/disable-hide-log-option-in-event-administration/223)

**Comment by bpedersen2** [comment]
A way to mark a field as 'needs data protection cleaning' in the regform setup would definitly be nice.

**Comment by stefanritt** [comment]
Probably it was me mentioning cookies. As it turns out, the GDPR affects the usage of cookies. Have a look here:

[How the GDPR affects cookie policies](https://www.itgovernance.eu/blog/en/how-the-gdpr-affects-cookie-policies/)

so I believe somethings needs to be done on the INDICO site.

**Comment by TimSmithCH** [comment]
@pferreir Indeed this applies to all relevant "personal data collection points" not just the registration form - I just used that (easy) example to kick start the list of changes!

It would indeed be good to make it clear that Indico itself only needs name/email to function, through the privacy statement template, that can then be extended by indico instance managers, and event event organisers

And yes I believe cookie usage is inevitably entangled in this...

**Comment by ThiefMaster** [comment]
Since we don't put personal data in cookies (or the session) until you log in: Would it be sufficient to have this as part of a privacy policy you need to agree to when logging in for the first time?
Like this it would not be as annoying to users (since everyone is used to "read" and accept one at some point when logging in to / signing up for a site the first time).

---

The link from @stefanritt ("when cookies can identify an individual via their device, it is considered personal data.") sounds a bit like any kind of session ID is problematic though while putting data like language/timezone directly in a cookie is fine. Quite annoying, since it means we'd need to add extra logic to put certain information directly in a cookie instead of using the server-side session. Or is it already enough to not *use* data for tracking/identifying users, even if you technically could?

---

On some other sites I read that with the new regulations you might only need consent for third-party cookies (can Piwik work without them btw?), but not for your own cookies that aren't used for tracking across sites...

**Comment by TimSmithCH** [comment]
On Cookies: indeed @ThiefMaster I believe you are right, the forthcoming ePrivacy directive is expected to relax the requirements, and so it won't be necessary to declare cookies that are normal and necessary for the functioning of the services, only tracking cookies. Hence the login-page approach seems excellent way to go

**Comment by vunnithan** [comment]
Dear All,
Since the deadline is approaching and it seems our IT administrators just woke up - is there someway to add a privacy statement that would conform (or at least go in the direction of conforming) to the GDPR ? many thanks for any pointers .. cheers Vikram

**Comment by ThiefMaster** [comment]
Right now you could probably use the "terms and conditions" for it as a workaround.

But we'll add an option to add a "Privacy" link next to the "terms and conditions" one 

---

> that would conform (or at least go in the direction of conforming) to the GDPR

Writing one is going to be up to you (or your organization's data protection officer / lawyer / whatever) for sure. We can't provide a default one since that would most likely open us / CERN to all kinds of liabilities. I think the original comment from @TimSmithCH about the default privacy statement was about CERN-hosted instances.

**Comment by vunnithan** [comment]
thanks .. yes, I guess the additional "Privacy" option next to terms and conditions would really help. Could these be on all pages / events etc ? 
Thanks for the clarification re default privacy statement. I will check and see what they have cooked up here. 
In the meanwhile, I was wondering, if someone could point me in the direction of how-to or where to change the "terms and conditions" .. I would be grateful for any pointers, many thanks Vikram

**Comment by ThiefMaster** [comment]
Yes, just like T&C this link will be visible on all pages.

You can change the conditions on this page:
Admin => Security => Legal/Disclaimers

**Comment by vunnithan** [comment]
:-) thanks

**Comment by u235axe** [comment]
Hi!
Related question: when creating a registration form for a conference, we'd like to link one or more custom GDPR documents in the registration form with a check box, that the user read the linked document and accepts the handling of her/his data. There seems to be no option for that now.

**Comment by bpedersen2** [comment]
Well, the checkbox is  available. What's a bit tricky is to add a link to the form, the description and free text fields currently support neither html nor markdown.
You could just add the text with a freetext field in the registrion form (Remember, GDPR requires easily understandable text anyway, so no need for excessive legalese their).

**Comment by u235axe** [comment]
Thanks for the answer, but we have no control over what our institute requires for compliance with GDPR, so now we have to somehow link like 3 pages of A4 text to the user.

A link type form element could really help us.

**Comment by pferreir** [comment]
> A link type form element could really help us.

https://github.com/indico/indico/issues/2168

**Comment by pferreir** [comment]
After a careful internal discussion (which included CERN's own DPO), we came up with a series of "mechanisms" we believe would cover the vast majority of potential GDPR issues while having limited impact in usability and the existing workflows.

Here is the list of points, in what we believe could be the order of implementation:

 1. Option to **link to a privacy policy** document on all pages (done ✅ );
 2. A user-accessible **list** of all **data Indico holds about a user**
    * *e.g. manager of "Event 1", speaker at "Contribution 2", owner of material "A.pdf", manager of "Category B"...*
 3. Introducing the concept of "**Data Controller**", the person/organization ultimately responsible for processing the data;
    * *It would by default be the person creating the event*;
    * *An "institution name" field should also be present, in case the controller is acting on behalf of a 
 4. **Clear warnings**:
     1. For **managers**: in management sections where personal data can be **gathered** (e.g. registration form) or **exposed** (e.g. lists of participants), linking to user documentation;
      2. For users: wherever a user is **expected to input personal data**, as well as the possibility to contact the Data Controller if they want it removed at some point;
company/organization*;
 5. **Registration data**:
     1. An interface that would allow organizers to **mark registration fields as personal data** and later **anonymize them** (manually or automatically), on a case by case basis or in bulk;
     2. A mechanism whereby users **consent explicitly** to having their name listed on the public **"List of Participants"** (only if the organizer wants the list to be visible);
     3. An additional checkbox in the registration form, confirming that the user has read the ToS and Privacy Policy, and that (s)he is not a minor;
 6. An interface that will allow an administrator to **anonymize an indico account**
    * *The user would be unlinked from any ACL entries and "EventPerson links". Those would be anonymized (name/affiliation replaced with a placeholder)*;

As you can see, the amount of changes is pretty extensive and will require considerable development time. We believe that by prioritizing user/transparency-related functionality we will cover the areas that are more likely to result in problems. Moreover, anonymization operations can be performed manually through `indico shell` anyway, in the worst case scenario.

We are still evaluating the resources we'd need to develop this, so the timeline will depend on how fast we can get those. As usual, help is very welcome. If you feel that you or your institution would be willing to lend a hand (and this is something that I believe will affect **the whole community**), don't hesitate to **let us know**!

**Comment by bpedersen2** [comment]
Part of the anoymization (with hard-coded fields for now...) is available via https://github.com/bpedersen2/indico-cron-advanced-cleaner

**Comment by dkfz2018** [comment]
A cronjob that deletes the hole event that was more than xx days ago (event date) should be presentated. Because you also have to delete all personal event informations regulary if you don´t get the consent to proceed with them for other stuff, it could be easier to jst delte the hole event rather than just anynomize specific fields.

**Comment by u235axe** [comment]
I disagree. We archive talk slides in the indico system, accessible from the timetable. Deleting the whole event would remove those archived slides too.

**Comment by pferreir** [comment]
> A cronjob that deletes the hole event that was more than xx days ago (event date) should be presentated.

As @u235axe wrote, that is pretty much contrary to the way 99% of organizations use Indico. If consent is needed, we should ask for it. And we should also automate the removal of unnecessary personal information after a given deadline.

**Comment by pferreir** [comment]
https://github.com/indico/indico/releases/tag/v3.2
